wolfssl/wolfcrypt/settings.h

```diff
@@ -4416,6 +4416,13 @@ extern void uITRON4_free(void *p) ;
     #endif
 #endif /* HAVE_ENTROPY_MEMUSE */
 
+/* If not server and not client, WOLFCRYPT_ONLY should be defined */
+#if defined(NO_WOLFSSL_CLIENT) && defined(NO_WOLFSSL_SERVER) && \
+   !defined(WOLFCRYPT_ONLY)
+    #error "ERROR: Both NO_WOLFSSL_CLIENT and NO_WOLFSSL_SERVER defined. "\
```

Comment:
`If TLS is enabled please make sure either client or server is enabled`

Reply:
yes, I can add that too. What's a good way to answer "_is TLS enabled_"?

`CONFIG_TLS_STACK_WOLFSSL` || `WOLFSSL_DTLS` || `HAVE_TLS_EXTENSIONS` || `WOLFSSL_TLS13` || `WOLFSSL_LEANTLS`  ?

Or is TLS _always_ enabled unless `NO_TLS` is found?

Reply:
Hi @dgarske I've introduced new logic to check both the `WOLFCRYPT_ONLY` and `NO_TLS`  along with the `NO_WOLFSSL_CLIENT` and `NO_WOLFSSL_SERVER` with appropriate error messages.

The `NO_TLS`  was a bit more interesting for me. Upon adding to my `user_setting.h` a variety of other compiler warnings were encountered. I appeased most of them, but perhaps all of the  

`#if !defined(WOLFSSL_NO_TLS12)` 

should be updated to 

`#if !defined(NO_TLS) && !defined(WOLFSSL_NO_TLS12)`